src/os.cpp

```diff
@@ -989,12 +989,29 @@ int os_self_exe_path(Buf *out_path) {
     }
 
 #elif defined(ZIG_OS_DARWIN)
+    // How long is the executable's path?
     uint32_t u32_len = 0;
     int ret1 = _NSGetExecutablePath(nullptr, &u32_len);
     assert(ret1 != 0);
-    buf_resize(out_path, u32_len);
-    int ret2 = _NSGetExecutablePath(buf_ptr(out_path), &u32_len);
+
+    // Make a buffer having room for the temp path.
+    Buf *tmp = buf_alloc_fixed(u32_len);
+
+    // Fill the executable path.
+    int ret2 = _NSGetExecutablePath(buf_ptr(tmp), &u32_len);
     assert(ret2 == 0);
+
+    // Resolve the real path from that.
+    buf_resize(out_path, PATH_MAX);
```

Comment:
Probably not relevant to Zig but in libuv we allocate 2*PATH_MAX to work around a bug in old versions of Apple's libc where the expanded path is sometimes > PATH_MAX.

Reply:
Thanks, I'll throw that in

Reply:
cd4676a2338430d9e424f297b8b576143c5be180